File: Leet Code Solutions/34. Find First and Last Position of Element in Sorted Array/PralinKhaira-FirstAndLastPosition.cs
public class Solution {
    public int[] SearchRange(int[] nums, int target) {
        int leftIndex = BinarySearch(nums, target, true);
        int rightIndex = BinarySearch(nums, target, false);

        if (leftIndex <= rightIndex) {
            return new int[] { leftIndex, rightIndex };
        } else {
            return new int[] { -1, -1 };
        }
    }

    private int BinarySearch(int[] nums, int target, bool findLeft) {
        int low = 0;
        int high = nums.Length - 1;
        int index = -1;

        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (nums[mid] == target) {
                index = mid;
                if (findLeft) {
                    high = mid - 1;
                } else {
                    low = mid + 1;
                }
            } else if (nums[mid] < target) {
                low = mid + 1;
            } else {
                high = mid - 1;
            }
        }

        return index;
    }
}

File: Leet Code Solutions/34. Find First and Last Position of Element in Sorted Array/PralinKhaira-FirstAndLastPosition.js
var searchRange = function(nums, target) {
    const findLeft = (nums, target) => {
        let left = 0;
        let right = nums.length - 1;
        let index = -1;

        while (left <= right) {
            let mid = Math.floor((left + right) / 2);
            if (nums[mid] === target) {
                index = mid;
                right = mid - 1;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return index;
    };

    const findRight = (nums, target) => {
        let left = 0;
        let right = nums.length - 1;
        let index = -1;

        while (left <= right) {
            let mid = Math.floor((left + right) / 2);
            if (nums[mid] === target) {
                index = mid;
                left = mid + 1;
            } else if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return index;
    };

    const leftIndex = findLeft(nums, target);
    const rightIndex = findRight(nums, target);

    if (leftIndex <= rightIndex) {
        return [leftIndex, rightIndex];
    } else {
        return [-1, -1];
    }
};